src/Parallel/DataCollector.h

```diff
@@ -0,0 +1,86 @@
+#ifndef SEISSOL_PARALLEL_DATACOLLECTOR_H
+#define SEISSOL_PARALLEL_DATACOLLECTOR_H
+
+#include <cstddef>
+#include <vector>
+
+#include "Kernels/precision.hpp"
+
+#ifdef ACL_DEVICE
+#include "device.h"
+#endif
+
+namespace seissol::parallel {
+
+class DataCollector {
+  public:
+  DataCollector(const std::vector<real*>& indexDataHost,
+                size_t elemSize,
+                bool hostAccessible = false)
+      : indexCount(indexDataHost.size()), indexDataHost(indexDataHost), elemSize(elemSize),
+        hostAccessible(hostAccessible) {
+#ifndef ACL_DEVICE
+    hostAccessible = true;
```

Comment:
should never happen, right?

Reply:
If we were to use this class in a non-GPU environment, then yes